### ISSUE TITLE
`melt` gives a result with that rows order that mismatches with pandas

### ISSUE THREAD

**Issue by dchigarev** [modin-project/modin#1793]
Current `melt` implementation shuffles rows order in the result, because we're applying function without information about the whole column axis. We should find a way how to quickly sort result to it matches pandas

**Comment by pyrito** [comment]
@dchigarev do you have a reproducer?

**Comment by dchigarev** [comment]
@pyrito yes, I've updated the [issues' header](https://github.com/modin-project/modin/issues/1793#issue-665139628) with the reproducer and added more details about the problem
